src/messaging/ExchangeMessageDispatch.cpp

```diff
@@ -49,8 +49,8 @@ CHIP_ERROR ExchangeMessageDispatch::SendMessage(SecureSessionHandle session, uin
     PayloadHeader payloadHeader;
     payloadHeader.SetExchangeID(exchangeId).SetMessageType(protocol, type).SetInitiator(isInitiator);
 
-    // If there is a pending acknowledgment piggyback it on this message.
-    if (reliableMessageContext->HasPeerRequestedAck())
+    // If there is a pending acknowledgment piggyback it on this message, and we have not send ACK before.
```

Comment:
```suggestion
    // If there is a pending acknowledgment, and the message had not
    // previously been acknowledged, piggyback an ACK on this message.
```

Reply:
The problem here is the MRP tick fire before the response message from the app, and it only clear the isAckPending flag, but not HasPeerRequestedAck flag. So the following response message  still piggy back the ack for the same message which has already been acked by a standalong ack message